[BUG] The migration of the CSb version 0.3.0 to 0.4.0 crashed using the mariadb version 10.4

## Description
The migration step db_service/models/historical_db.go `ServiceInstanceDetailsV3 ` present in the releas/tag 0.4.0 to expends the size of the workspace column to handle the deployment crash the application using the mariadb.

### Mariadb version
```
+--------------------------------------------+
| VERSION()                                  |
+--------------------------------------------+
| 10.4.10-MariaDB-1:10.4.10+maria~buster-log |
+--------------------------------------------+
1 row in set (0.001 sec)
```

## Expected Behavior
The application should work properly even if the update of the row is not possible.

## Actual Behavior
The application does not start with the following error message:
```
{"timestamp":"1635517167.577763557","source":"cloud-service-broker","message":"cloud-service-broker.Connecting to MySQL Database","log_level":1,"data":{"host":"csb_mariadb","name":"csb","port":"3306"}}

2021/10/29 14:19:27 /root/go/pkg/mod/gorm.io/driver/mysql@v1.1.2/migrator.go:82 Error 1406: Data too long for column 'workspace' at row 8
[131.510ms] [rows:0] ALTER TABLE `terraform_deployments` MODIFY COLUMN `workspace` blob
panic: Error migrating database: Error 1406: Data too long for column 'workspace' at row 8

goroutine 1 [running]:
gitlab.tech.orange/group-fpc/forks/cloud-service-broker/db_service.New.func1()
        /app/db_service/db_service.go:37 +0xb4
sync.(*Once).doSlow(0xc000319780, 0x453112)
        /usr/local/go/src/sync/once.go:68 +0xd2
sync.(*Once).Do(...)
        /usr/local/go/src/sync/once.go:59
gitlab.tech.orange/group-fpc/forks/cloud-service-broker/db_service.New({0x1300ac8, 0xc000316ba0})
        /app/db_service/db_service.go:34 +0x5e
gitlab.tech.orange/group-fpc/forks/cloud-service-broker/cmd.serve()
        /app/cmd/serve.go:84 +0x4f
gitlab.tech.orange/group-fpc/forks/cloud-service-broker/cmd.init.5.func1(0xc000209400, {0x10f8652, 0x0, 0x0})
        /app/cmd/serve.go:62 +0x17
github.com/spf13/cobra.(*Command).execute(0xc000209400, {0x1a4fb28, 0x0, 0x0})
        /root/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:860 +0x5f8
github.com/spf13/cobra.(*Command).ExecuteC(0x1985100)
        /root/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:974 +0x3bc
github.com/spf13/cobra.(*Command).Execute(...)
        /root/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:902
gitlab.tech.orange/group-fpc/forks/cloud-service-broker/cmd.Execute()
        /app/cmd/root.go:43 +0x25
main.main()
        /app/main.go:22 +0x17
```

## The root cause:
The root cause of this issue is due to the migration from mediumtext to blob showed bellow:

### In db_service/migrations.go
```
migrations[13] = func() error {
		return db.Migrator().AlterColumn(&models.TerraformDeploymentV3{}, "workspace")
	}
```

## In db_service/models/historical_db.go where be
```
// TerraformDeploymentV2 expands the size of the Workspace column to handle deployments where the
// Terraform workspace is greater than 64K. (mediumtext allows for workspaces up
// to 16384K.)
type TerraformDeploymentV2 struct {
	ID        string `gorm:"primary_key;type:varchar(1024)"`
	CreatedAt time.Time
	UpdatedAt time.Time
	DeletedAt *time.Time

	// Workspace contains a JSON serialized version of the Terraform workspace.
	Workspace string `gorm:"type:mediumtext"`

	// LastOperationType describes the last operation being performed on the resource.
	LastOperationType string

	// LastOperationState holds one of the following strings "in progress", "succeeded", "failed".
	// These mirror the OSB API.
	LastOperationState string

	// LastOperationMessage is a description that can be passed back to the user.
	LastOperationMessage string `gorm:"type:text"`
}
```
```
// TerraformDeploymentV3 expands the size of the Workspace column to handle deployments where the
// Terraform workspace is greater than 64K. (mediumtext allows for workspaces up
// to 16384K.)
type TerraformDeploymentV3 struct {
	ID        string `gorm:"primary_key;type:varchar(1024)"`
	CreatedAt time.Time
	UpdatedAt time.Time
	DeletedAt *time.Time

	// Workspace contains a JSON serialized version of the Terraform workspace.
	Workspace []byte `gorm:"type:blob"`

	// LastOperationType describes the last operation being performed on the resource.
	LastOperationType string

	// LastOperationState holds one of the following strings "in progress", "succeeded", "failed".
	// These mirror the OSB API.
	LastOperationState string

	// LastOperationMessage is a description that can be passed back to the user.
	LastOperationMessage string `gorm:"type:text"`
}
```
## Possible Fix
Setting up the TerraformDeploymentV3 to TerraformDeploymentV2 that still use mediumtext for the workspace column but I assume that we have to check if the update is possible instead.
### In db_service/migrations.go
```
migrations[13] = func() error {
		return db.Migrator().AlterColumn(&models.TerraformDeploymentV2{}, "workspace")
	}
```

## Steps to Reproduce
1. Initiate a mariadb database 
2. Run the Cloud service broker version 0.3.0 with this mariadb datase
2. Update the cloud service broker with the release 0.4.0

## Context
As service provide at Orange, I use this broker to provide all our services throught terraform as we try to promote this way to create and consume services.

## Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->
* Version used: 0.3.0 - 0.4.0 
* Operating System and version (desktop): Docker Ubuntu 20.04 
* Link to your project (if public): https://github.com/orange-cloudfoundry/cloud-service-broker
* Platform (Azure/AWS/GCP): Vmware

--- a/db_service/migrations.go
+++ b/db_service/migrations.go
@@ -22,7 +22,7 @@ import (
 	"gorm.io/gorm"
 )
 
-const numMigrations = 14
+const numMigrations = 15
 
 // RunMigrations runs schema migrations on the provided service broker database to get it up to date
 func RunMigrations(db *gorm.DB) error {
@@ -113,6 +113,10 @@ func RunMigrations(db *gorm.DB) error {
 
 	migrations[13] = func() error {
 		return db.Migrator().AlterColumn(&models.TerraformDeploymentV3{}, "workspace")
+	}
+
+	migrations[14] = func() error {
+		return db.Migrator().AlterColumn(&models.TerraformDeploymentV4{}, "workspace")
 	}
 
 	var lastMigrationNumber = -1

--- a/db_service/models/db.go
+++ b/db_service/models/db.go
@@ -145,7 +145,7 @@ type CloudOperation CloudOperationV1
 
 // TerraformDeployment holds Terraform state and plan information for resources
 // that use that execution system.
-type TerraformDeployment TerraformDeploymentV3
+type TerraformDeployment TerraformDeploymentV4
 
 func (t *TerraformDeployment) SetWorkspace(value string) error {
 	encrypted, err := encryptorInstance.Encrypt([]byte(value))

--- a/db_service/models/historical_db.go
+++ b/db_service/models/historical_db.go
@@ -365,6 +365,36 @@ func (TerraformDeploymentV3) TableName() string {
 	return "terraform_deployments"
 }
 
+// TerraformDeploymentV4 expands the size of the Workspace column to handle deployments where the
+// Terraform workspace is greater than 64K. (mediumblob allows for workspaces up
+// to 16384K.)
+type TerraformDeploymentV4 struct {
+	ID        string `gorm:"primary_key;type:varchar(1024)"`
+	CreatedAt time.Time
+	UpdatedAt time.Time
+	DeletedAt *time.Time
+
+	// Workspace contains a JSON serialized version of the Terraform workspace.
+	Workspace []byte `gorm:"type:mediumblob"`
+
+	// LastOperationType describes the last operation being performed on the resource.
+	LastOperationType string
+
+	// LastOperationState holds one of the following strings "in progress", "succeeded", "failed".
+	// These mirror the OSB API.
+	LastOperationState string
+
+	// LastOperationMessage is a description that can be passed back to the user.
+	LastOperationMessage string `gorm:"type:text"`
+}
+
+// TableName returns a consistent table name for
+// gorm so multiple structs from different versions of the database all operate
+// on the same table.
+func (TerraformDeploymentV4) TableName() string {
+	return "terraform_deployments"
+}
+
 // PasswordMetadataV1 contains information about the passwords, but never the
 // passwords themselves
 type PasswordMetadataV1 struct {

